src/defect/DefectService.ts

```diff
@@ -0,0 +1,223 @@
+				const { imNumber, imDescription, imDescriptionWelsh, items } = defect;
+				if (defect.items !== undefined && defect.items.length !== 0) {
+					// go through each item of defect
+					items.forEach((item: IItem) => {
+						const { itemNumber, itemDescription, itemDescriptionWelsh, deficiencies } = item;
+						if (item.deficiencies !== undefined && item.deficiencies.length !== 0) {
+							// go through each deficiency and push to flatDefects array
+							deficiencies.forEach((deficiency: IDefectChild) => {
+								const { ref, deficiencyText, deficiencyTextWelsh, forVehicleType } = deficiency;
+								const lowLevelDeficiency: IFlatDefect = {
+									imNumber,
+									imDescription,
+									imDescriptionWelsh,
+									itemNumber,
+									itemDescription,
+									itemDescriptionWelsh,
+									ref,
+									deficiencyText,
+									deficiencyTextWelsh,
+									forVehicleType,
+								};
+								flatDefects.push(lowLevelDeficiency);
+							});
+						}
+					});
+				}
+			});
+		} catch (e) {
+			console.error(`Error flattening defects: ${e}`);
+		}
+		return flatDefects;
+	}
+
+	/**
+	 * Returns a formatted string containing data about a given defect
+	 * @param defect - defect for which to generate the formatted string
+	 */
+	public formatDefect(defect: any) {
+		const toUpperFirstLetter: any = (word: string) => word.charAt(0).toUpperCase() + word.slice(1);
+
+		let defectString = `${defect.deficiencyRef} ${defect.itemDescription}`;
+
+		if (defect.deficiencyText) {
+			defectString += ` ${defect.deficiencyText}`;
+		}
+
+		if (defect.additionalInformation.location) {
+			Object.keys(defect.additionalInformation.location).forEach((location: string, index: number, array: string[]) => {
+				if (defect.additionalInformation.location[location]) {
+					switch (location) {
+						case 'rowNumber':
+							defectString += ` Rows: ${defect.additionalInformation.location.rowNumber}.`;
+							break;
+						case 'seatNumber':
+							defectString += ` Seats: ${defect.additionalInformation.location.seatNumber}.`;
+							break;
+						case 'axleNumber':
+							defectString += ` Axles: ${defect.additionalInformation.location.axleNumber}.`;
+							break;
+						default:
+							defectString += ` ${toUpperFirstLetter(defect.additionalInformation.location[location])}`;
+							break;
+					}
+				}
+
+				if (index === array.length - 1) {
+					defectString += `.`;
+				}
+			});
+		}
+
+		if (defect.additionalInformation.notes) {
+			defectString += ` ${defect.additionalInformation.notes}`;
+		}
+
+		return defectString;
+	}
+
+	/**
+	 * Returns a formatted welsh string containing data about a given defect
+	 * @param defect - the defect for which to generate the formatted welsh string
+	 * @param vehicleType - the vehicle type from the test result
+	 * @param flattenedDefects - the list of flattened defects
+	 */
+	public formatDefectWelsh(defect: any, vehicleType: any, flattenedDefects: IFlatDefect[]) {
```

Comment:
I'd be inclined to say to merge the defect formatting into one function given that they're both functionally very similar to each other

Reply:
Yep agree with this, same as the previous comments. I'm trying to constraint these PRs to structural changes, functional changes will be easier to review in isolation in a follow up PR.